client/components/RegisterPage.tsx

```diff
@@ -1,64 +1,133 @@
 import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { 
+  TextField, 
+  Box, 
+  Typography, 
+  InputAdornment,
+  IconButton,
+  Button
+} from "@mui/material";
+import {
+  Visibility,
+  VisibilityOff
+} from "@mui/icons-material";
 import { useRegisterMutation } from "../features/api";
 
 const RegisterPage = () => {
   const [register, { isLoading, isError, data}] = useRegisterMutation();
 
+  const navigate = useNavigate();
+
   const [email, setEmail] = useState("");
   const [username, setUsername] = useState("");
   const [password, setPassword] = useState("");
   const [confirmPassword, setConfirmPassword] = useState("");
+  const [showPassword, setShowPassword] = useState(false);
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false);
+
+  const handleClickShowPassword = () => setShowPassword((show) => !show);
+  const handleMouseDownPassword = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+  };
+
+  const handleClickShowConfirmPassword = () => setShowConfirmPassword((show) => !show);
+  const handleMouseDownConfirmPassword = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+  };  
 
   const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
-    register({ email, username, password });
+    if (password === confirmPassword) {
+      register({ email, username, password });
+      navigate("/")
+    } else {
+      alert("Password confirmation does not match");
+    }
   };
 
   return (
-      <form onSubmit={handleSubmit}>
-        <h1>Register</h1>
-        {isError && <p>Oops, there was an error creating an account. Try again?</p>}
+      <Box 
+        component="form"
+        onSubmit={handleSubmit}
+        autoComplete="off"
```

Comment:
same note as with login - confirming that you want to turn off this default browser behavior and why